dts/binding-template.yaml

```diff
@@ -125,6 +126,15 @@ sub-node:
 #   foo = <&label1 &label2>, <&label3 &label4>; // Okay for 'type: phandles'
 #   foo = <&label1 1 2>, <&label2 3 4>;         // Okay for 'type: phandle-array'
 #   etc.
+#
+# The optional 'default:' setting gives a value that will be used if the
+# property is missing from the device tree node. If 'default: <default>' is
+# given for a property <prop> and <prop> is missing, then the output will be as
+# if '<prop> = <default>' had appeared (except YAML data types are used for the
+# default value). Note that it only makes sense to combine 'default:' with
+# 'required: false'.
```

Comment:
Do we error check that `default:` is only set for `required: false`?

Reply:
I'll add a check.

Reply:
Hmm... wonder if this might be too strict though. If a binding changes a property from an included binding that happens to have has a `default:` from `required: false` to `required: true`, then it might be bad to flag that as an error.

I'll leave it out unless you think it should be flagged.

Reply:
if we inherit in such a case, then `required` goes from `false` -> `true`, then the `default` should not propagate.

I'd rather us add the check and see if such a case comes up.

Reply:
Alright, I'll add the check.

I'll skip stripping the `default` for now. Could see later if it's actually needed.

Reply:
Added check and updated documentation.

Reply:
At one point I thought I suggested that `default` be used only if `required`.  It doesn't make sense to have `optional` with a default because the motivation of `optional` is to indicate that something may not be provided at all, in which case the code probably does something special.

Reply:
@pabigot not sure I follow.